src/lib.rs

```diff
@@ -301,3 +303,36 @@ fn display_columns(bytes: &[u8]) -> usize {
                   .fold(0, |sum, width| sum + width),
     }
 }
+
+#[cfg(feature = "ansi_formatting")]
+fn display_columns(bytes: &[u8]) -> usize {
+    use unicode_width::UnicodeWidthChar;
+
+    // If we have a Unicode string, then attempt to guess the number of
+    // *display* columns used.
+    match str::from_utf8(bytes) {
+        Err(_) => bytes.len(),
+        Ok(s) => {
+            if let Some(stripped_s) = strip_formatting(s) {
+                stripped_s.chars()
+                    .map(|c| UnicodeWidthChar::width(c).unwrap_or(0))
+                    .fold(0, |sum, width| sum + width)
+            } else {
+                s.chars()
+                    .map(|c| UnicodeWidthChar::width(c).unwrap_or(0))
+                    .fold(0, |sum, width| sum + width)
+            }
+        }
+    }
+}
+
+#[cfg(feature = "ansi_formatting")]
+fn strip_formatting(input: &str) -> Option<String> {
+    // check if the input actually contains ANSI escape codes
+    // to avoid unnecessary allocations
+    if input.contains("\x1b[") {
+        let re = regex::Regex::new("\x1B\\[.+?m").unwrap();
```

Comment:
Hmm. So this is going to re-compile the regex on every single call to `strip_formatting`. The easiest way to work around this is to probably use the `lazy_static!` crate. I think something like this should work:

```rust
fn strip_formatting(input: &str) -> Option<String> {
    lazy_static! {
        static ref re: regex::Regex = Regex::new("\x1B\\[.+?m").unwrap();
    }
}
```